src/pages/index.js

```diff
@@ -0,0 +1,327 @@
+    } else {
+      api
+        .changeLikeStatus(data._id)
+        .then((res) => {
+          cardLikeBtn.classList.add("card__like-btn_liked");
+        })
+        .catch(console.error);
+    }
+  });
+
+  cardDeleteBtn.addEventListener("click", () => {
+    openModal(deleteModal);
+    deleteModal.dataset.cardId = data._id;
+
+    cardElement.dataset.cardId = data._id;
+  });
+
+  return cardElement;
+}
+
+function handleDeleteConfirmation() {
+  const cardId = deleteModal.dataset.cardId;
+  const cardElement = document.querySelector(`.card[data-card-id="${cardId}"]`);
+
+  const deleteBtn = document.querySelector(".modal__submit-btn_type_delete");
+  setButtonText(deleteBtn, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      cardElement.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(deleteBtn, false, "Delete", "Deleting...");
+    });
+}
+
+const deleteForm = deleteModal.querySelector(".modal__form");
+deleteForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  handleDeleteConfirmation();
+});
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .addCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      evt.target.reset();
+      disableButton(cardSubmitBtn, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscape);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscape);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editAvatarInfo({ avatar: avatarLinkInput.value })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      setButtonText(submitBtn, false);
+      avatarForm.reset();
+      disableButton(avatarSubmitBtn, settings);
```

Comment:
```js
      avatarForm.reset();
      disableButton(avatarSubmitBtn, settings);
```

These 2 lines of code should be in line 263 in `.then`



```js

    .then((data) => {
      profileAvatar.src = data.avatar;
      closeModal(avatarModal);
      avatarForm.reset();  // here
      disableButton(avatarSubmitBtn, settings); // here
    })
    .catch((err) => {
      console.log(err);
    })
    .finally(() => {
      setButtonText(submitBtn, false);
    });
```

Reply:
You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.